core/src/test/java/com/sequenceiq/cloudbreak/service/multiaz/MultiAzCalculatorServiceTest.java

```diff
@@ -0,0 +1,156 @@
+
+    @InjectMocks
+    private MultiAzCalculatorService underTest;
+
+    @Test
+    public void testPrepareSubnetAzMap() {
+        DetailedEnvironmentResponse detailedEnvironmentResponse = new DetailedEnvironmentResponse();
+        EnvironmentNetworkResponse environmentNetworkResponse = new EnvironmentNetworkResponse();
+        environmentNetworkResponse.setSubnetMetas(Map.of(
+                cloudSubnetName(1), cloudSubnet(1),
+                cloudSubnetName(2), cloudSubnet(2)
+        ));
+        detailedEnvironmentResponse.setNetwork(environmentNetworkResponse);
+
+        Map<String, String> result = underTest.prepareSubnetAzMap(detailedEnvironmentResponse);
+        Assert.assertEquals(result.size(), 4);
+        Assert.assertEquals(result.get(cloudSubnetName(1)), cloudSubnetAz(1));
+        Assert.assertEquals(result.get(cloudSubnetName(2)), cloudSubnetAz(2));
+    }
+
+    static Object[][] testSubnetDistributionData() {
+        return new Object[][]{
+                { CloudPlatform.AWS,    1,  99,  List.of(99),               true   },
+                { CloudPlatform.AWS,    3,  99,  List.of(33, 33, 33),       true   },
+                { CloudPlatform.AWS,    3,  60,  List.of(20, 20, 20),       true   },
+                { CloudPlatform.AWS,    3,  59,  List.of(20, 20, 19),       true   },
+                { CloudPlatform.AWS,    3,  58,  List.of(20, 19, 19),       true   },
+                { CloudPlatform.AWS,    3,  57,  List.of(19, 19, 19),       true   },
+                { CloudPlatform.AWS,    3,  56,  List.of(19, 19, 18),       true   },
+                { CloudPlatform.AWS,    4,  60,  List.of(15, 15, 15, 15),   true   },
+                { CloudPlatform.AWS,    4,  59,  List.of(15, 15, 15, 14),   true   },
+                { CloudPlatform.AWS,    4,  58,  List.of(15, 15, 14, 14),   true   },
+                { CloudPlatform.AWS,    4,  57,  List.of(15, 14, 14, 14),   true   },
+                { CloudPlatform.AWS,    4,  56,  List.of(14, 14, 14, 14),   true   },
+                { CloudPlatform.YARN,   0,  99,  List.of(),                 false  },
+        };
+    }
+
+    @ParameterizedTest(name = "testSubnetDistribution with {0} platform when {1} should result as {2}")
+    @MethodSource("testSubnetDistributionData")
+    public void testSubnetDistribution(CloudPlatform cloudPlatform, int subnetCount, int instanceCount,
+        List<Integer> expected, boolean supported) {
+        if (supported) {
+            when(multiAzValidator.supportedForInstanceMetadataGeneration(any(InstanceGroup.class))).thenReturn(true);
+        }
+        int subnetNumber = subnetCount;
+        InstanceGroup instanceGroup = instanceGroup(cloudPlatform, instanceCount, subnetNumber);
+
+        underTest.calculateByRoundRobin(
+                subnetAzPairs(subnetNumber),
+                instanceGroup);
+
+        List<Integer> actual = new ArrayList<>();
+        for (int i = 0; i < subnetNumber; i++) {
+            int finalI = i;
```

Comment:
What is the reason for this extra variable?

Reply:
you can not use the "i" in the stream

Reply:
Ah, right, sorry.